File: json_to_flutter/models/submodels/radius.dart
import '../../functions/util.dart';

class Radius {
  double topLeft;
  double topRight;
  double bottomRight;
  double bottomLeft;

  Radius({this.topLeft, this.topRight, this.bottomRight, this.bottomLeft});

  Radius.fromJson(Map<String, dynamic> json) {
    topLeft = fixDouble(json['topLeft']);
    topRight = fixDouble(json['topRight']);
    bottomRight = fixDouble(json['bottomRight']);
    bottomLeft = fixDouble(json['bottomLeft']);
  }
  bool isCircular() {
    return topLeft == topRight &&
        topRight == bottomRight &&
        bottomRight == bottomLeft;
  }
}

File: src/functions/widget/models/submodels/radius.js
const { fixDouble } = require("../../util");

class Radius {
  constructor(json) {
    this.topLeft = fixDouble(json['topLeft']);
    this.topRight = fixDouble(json['topRight']);
    this.bottomRight = fixDouble(json['bottomRight']);
    this.bottomLeft = fixDouble(json['bottomLeft']);
  }

  isCircular() {
    return this.topLeft == this.topRight &&
      this.topRight == this.bottomRight &&
      this.bottomRight == this.bottomLeft;
  }
}

module.exports = { Radius };